tests/testutils/goroutineassistant.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutils
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+)
+
+type GoroutineAssistant struct {
+	wait chan bool
+	fail chan bool
+	l    sync.Mutex
+	wg   sync.WaitGroup
+	t    *testing.T
+	s    string
+}
```

Comment:
I'm afraid this will not end cleanly because it will either block forever on `a.wg.Wait()` or on the `<- a.fail`. Either all goroutines end cleanly and do not write to the fail channel, or at least one goroutine fails and will not be able to get to the `wg.Done()` call.

Unfortunately the situation is more complex than it might look on the first (or second) sight.